File: treebolicClientsIface/src/main/java/org/treebolic/clients/iface/IModelListener.java
/*
 * Copyright (c) Treebolic 2019. Bernard Bou <1313ou@gmail.com>
 */

package org.treebolic.clients.iface;

import treebolic.model.Model;

/**
 * Model consumer interface
 *
 * @author Bernard Bou
 */
public interface IModelListener
{
	/**
	 * Model available callback
	 *
	 * @param model     model
	 * @param urlScheme url scheme
	 */
	void onModel(Model model, String urlScheme);
}

File: treebolicClientsIface/src/main/java/org/treebolic/clients/iface/IModelListener.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.treebolic.clients.iface

import treebolic.model.Model

/**
 * Model consumer interface
 *
 * @author Bernard Bou
 */
interface IModelListener {

    /**
     * Model available callback
     *
     * @param model     model
     * @param modelUrlScheme url scheme
     */
    fun onModel(model: Model?, modelUrlScheme: String?)
}
